daemon/api.go

```diff
@@ -2352,6 +2360,80 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
 	}, nil)
 }
 
+type postUserRemoveData struct {
+	Username string `json:"username"`
+}
+
+func postRemoveUser(c *Command, r *http.Request, user *auth.UserState) Response {
+	_, uid, _, err := postRemoveUserUcrednetGet(r.RemoteAddr)
```

Comment:
This is neat, not even decoding the rest of the request unless the calling user is root.

Reply:
Yes, it's the same pattern used in user creation.

Reply:
This check is unnecessary: the function will not be called unless `uid` is 0 (you can follow the logic [here](https://github.com/snapcore/snapd/blob/master/daemon/daemon.go#L116))
Yes, create-user also does it. It also doesn't need to.

The only case this would be catching is when the user is not root, but has authenticated via macaroons. If w realy want to have a 'you really need to be root _right now_ to do this' option (@mvo5 or maybe @pedronis might have opinions here), the right way is to add another flag (next to `GuestOK`), and use that from `canAccess`.